components/world-4/breeding/pets.tsx

```diff
@@ -0,0 +1,522 @@
+                if (pet2.shinyLevel >= 20) {
+                    return -1;
+                }
+                // Else sort by time till next level.
+                return (pet1.getNextShinyGoal() - pet1.shinyProgress) > (pet2.getNextShinyGoal() - pet2.shinyProgress) ? 1 : -1;
+            }
+
+            // Sort by level sort function
+            function sortByLevel(pet1: Pet, pet2: Pet) {
+                if (pet1.shinyLevel == pet2.shinyLevel) {
+                    // If level is equal, sort by time until next level
+                    return sortByTime(pet1, pet2);
+                } else {
+                    // if level isn't equal, just sort by level
+                    return pet1.shinyLevel > pet2.shinyLevel ? -1 : 1;
+                }
+            }
+
+            switch (sort) {
+                case "Level":
+                    return sortByLevel(pet1, pet2);
+                case "Least Time to Next Level":
+                    return sortByTime(pet1, pet2);
+                default:
+                    return indexSort;
+            }
+        })
+    }, [breeding, sort, filter])
+
+    if (!breeding) {
+        return (
+            <Box>
+                Still loading
+            </Box>
+        )
+    }
+
+    return (
+        <Box gap="small">
+            <Text>Shiny Bonuses</Text>
+            <Box direction="row" gap="medium" margin={{bottom: "small"}}>
+                <Select size="small"
+                    placeholder="Sort by"
+                    clear
+                    value={sort}
+                    options={sortOptions}
+                    onChange={({ value: nextValue }) => { setSort(nextValue); }}
+                />
+                <SelectMultiple
+                    size="small"
+                    placeholder="Filter by"
+                    searchPlaceholder="Search bonuses"
+                    value={filter}
+                    options={currentFilterOptions}
+                    onChange={({ value: nextValue }) => { setFilter(nextValue); }}
+                    onSearch={text => {
+                        // The line below escapes regular expression special characters:
+                        // [ \ ^ $ . | ? * + ( )
+                        const escapedText = text.replace(/[-\\^$*+?.()|[\]{}]/g, '\\$&');
+
+                        // Create the regular expression with modified value which
+                        // handles escaping special characters. Without escaping special
+                        // characters, errors will appear in the console
+                        const exp = new RegExp(escapedText, 'i');
+                        setCurrentFilterOptions(allFilterOptions.filter(o => exp.test(o)));
+                    }}
+                    onClose={() => setCurrentFilterOptions(allFilterOptions)}
+                />
+            </Box>
+            <Text size='medium'>Shiny speed multiplier : {nFormatter(breeding.getShinySpeed(starSignEquipped, silkRodeChip))}</Text>
```

Comment:
Please use `TextAndLabel` component for this,

Reply:
Done